component/component_provider.go

```diff
@@ -143,6 +146,7 @@ func (info *Info) MarshalJSON() ([]byte, error) {
 		Arguments: arguments,
 		Exports:   exports,
 		DebugInfo: debugInfo,
+		ModuleIDs: info.ModuleIDs,
```

Comment:
Thoughts on changing this name to something like `Child(Sub)ModuleIDs`? This feels like you have to think what ModuleID and ModuleIDs are in this context.

Reply:
I went for `CreatedModuleIDs` to make it clearer that they're modules created by the component, LMK if you think ChildModuleIDs/SubModuleIDs is preferable.